index.html

```diff
@@ -428,6 +435,56 @@ <h3>Constructor</h3>
           </li>
         </ol>
       </section>
+      <section data-dfn-for="WakeLock">
+        <h3><dfn>requestPermission()</dfn> static method</h3>
+        <p>
+          The <a>requestPermission()</a> method, when invoked, MUST run the
+           following steps:
+        </p>
+        <ol class="algorithm">
+          <li>
+            Let |promise| be a new <a>Promise</a> object.
+          </li>
+          <li>
+            Let |type| be the first argument.
+          </li>
+          <li>
+            Let |permissionDesc| be the following
+            <a>permission descriptor type</a>:
+            <pre>
+              {
+                <a data-lt="permission name">name</a>: "wake-lock",
+                type: <var>type</var>
+              }
+            </pre>
+          </li>
+          <li>
+            Return |promise| and run the following steps <a>in parallel</a>:
+            <ol>
+              <li>
+                Let |resultPromise| be the result of running
+                <a>query a permission</a> with |permissionDesc|.
```

Comment:
Instead of using the "query a permission" algorithm and restating a lot of the same logic from above can we reuse the "obtain permission" algorithm?

Reply:
That makes things worse :-) I tried